src/api/entities/confidential/ConfidentialAsset/index.ts

```diff
@@ -0,0 +1,99 @@
+import { Context, Entity, Identity } from '~/internal';
+import { ConfidentialAssetDetails } from '~/types';
+import {
+  bytesToString,
+  identityIdToString,
+  serializeConfidentialAssetId,
+  tickerToString,
+  u128ToBigNumber,
+} from '~/utils/conversion';
+import { assertCaAssetValid } from '~/utils/internal';
+
+/**
+ * Properties that uniquely identify a ConfidentialAsset
+ */
+export interface UniqueIdentifiers {
+  /**
+   * GUID of the asset
+   *
+   * @note the value can either be a valid guid like `76702175-d8cb-e3a5-5a19-734433351e26` or can be a string representing the guid without the `-` like `76702175d8cbe3a55a19734433351e26`
```

Comment:
After reading through the PR, I am thinking these should just be plain hex values.

These should be chain generated, and I am assuming all 128 bits are random. We shouldn't insert the `-` and call them GUIDs or UUIDs unless they conform to the actual specs. Both of those terms refer to something well defined, and its misleading to label random bits and them.

https://en.wikipedia.org/wiki/Universally_unique_identifier#Versions